Formula/gcc.rb

```diff
@@ -111,35 +80,21 @@ def install
       "--with-mpc=#{Formula["libmpc"].opt_prefix}",
       "--with-isl=#{Formula["isl"].opt_prefix}",
       "--with-system-zlib",
-      "--enable-stage1-checking",
       "--enable-checking=release",
-      "--enable-lto",
-      # Use 'bootstrap-debug' build configuration to force stripping of object
-      # files prior to comparison during bootstrap (broken by Xcode 6.3).
-      "--with-build-config=bootstrap-debug",
-      "--disable-werror",
       "--with-pkgversion=Homebrew GCC #{pkg_version} #{build.used_options*" "}".strip,
       "--with-bugurl=https://github.com/Homebrew/homebrew-core/issues",
     ]
 
-    # The pre-Mavericks toolchain requires the older DWARF-2 debugging data
-    # format to avoid failure during the stage 3 comparison of object files.
-    # See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=45248
-    args << "--with-dwarf2" if MacOS.version <= :mountain_lion
-
-    args << "--disable-nls" if build.without? "nls"
-
-    if build.with?("java") || build.with?("all-languages")
-      args << "--with-ecj-jar=#{Formula["ecj"].opt_share}/java/ecj.jar"
-    end
+    # Use -headerpad_max_install_names in the build,
+    # otherwise lto1 load commands cannot be edited on El Capitan
+    patch :DATA if MacOS.version == :el_capitan
```

Comment:
It would be good to
1. find out if we can get it to use the superenv shim
2. put this patch in the formula-patches repository if we're going to have to carry it around for a while
3. create upstream issue about the trouble using the `--with-boot-ldflags` option successfully